Density/app/src/main/java/org/cornelldti/density/density/FacilitiesListAdapter.kt

```diff
@@ -94,19 +92,23 @@ class FacilitiesListAdapter(data: ArrayList<Facility>) : RecyclerView.Adapter<Fa
     override fun getFilter(): Filter {
         return object : Filter() {
             override fun performFiltering(charSequence: CharSequence): FilterResults {
-                val charString = charSequence.toString().toLowerCase()
-                if (!charString.isEmpty()) {
-                    val filteredList = ArrayList<Facility>()
-                    for (f in facilities!!) {
-                        if (f.name!!.toLowerCase().contains(charString.toLowerCase())) {
-                            filteredList.add(f)
+                val charString = charSequence.toString().toLowerCase(Locale.US)
+                when(charString.isNotEmpty()) {
```

Comment:
This seems like a misuse of when expression. If the only two cases are `true` and `false`, you should just use if.

Reply:
Android Studio was warning me about being able to lift out the if statement?

Reply:
I'm thinking about
```kotlin
val charString = charSequence.toString().toLowerCase(Locale.US)
val filterResults = FilterResults()
dataSet = if (charString.isNotEmpty()) {
    val filteredList = ArrayList<Facility>()
    for (f in facilities!!) {
        if (f.name!!.toLowerCase(Locale.US).contains(charString.toLowerCase(Locale.US))) {
            filteredList.add(f)
        }
    }
    Collections.sort(filteredList, SEARCH_SORT.apply(charString))
    filteredList
} else {
    val mutFacilities = facilities!!.toMutableList()
    mutFacilities.sort()
    mutFacilities
}
```

Reply:
Should this be lifted out as well? This one is assigned to a value, so I guess changing it to a `when` makes less sense.

Reply:
I don't have much context on what's the purpose of database.
In terms of the kotlin code, there exists a way to turn this entire function body into a single expression through extension functions and scope functions (https://kotlinlang.org/docs/reference/scope-functions.html), but I don't think it greatly improves readability. Personally I think the if structure is good enough.